pkg/resourcecollector/resourcecollector.go

```diff
@@ -576,6 +579,10 @@ func (r *ResourceCollector) objectToBeCollected(
 		return false, err
 	}
 
+	if exclude, err := r.excludeResource(object, excludeResourceSelectors, namespace); err != nil || exclude {
+		return false, err
```

Comment:
Can you wrap the error here, so that it is easier to understand what failed while  collecting the object?
```
return false, fmt.Errorf("failed to determine if object needs to be excluded: %v", err)
```

Reply:
There are two conditions being checked inside if. Updating it in the following way
```     
 exclude, err := r.excludeResource(object, excludeResourceSelectors, namespace)
       if err != nil {
              return false, fmt.Errorf("failed to determine if object needs to be excluded: %v", err)
       }
       if exclude {
               return false, nil
       }
```